### PR TITLE
TrustyAI CounterfactualExplainer tests timeouts

### PR DESCRIPTION
**Pull request by spolti**
Many thanks for submitting your Pull Request :heart:!

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](https://github.com/trustyai-explainability/trustyai-explainability/blob/main/CONTRIBUTING.md)
- [ ] Pull Request title is properly formatted: `FAI-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting main: `[0.3.x] FAI-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Review by ruivieira — COMMENTED** [review]
@spolti thanks for the PR.

LGTM. As we discussed elsewhere, I think the only change could be to optionally pass an ExecutorService to the CF Explainer, with a default if none provided.
That way we could supply our own when using it with Quarkus, and use the default in other scenarios (e.g. from the Python bindings).

File: explainability-core/src/test/java/org/kie/trustyai/explainability/local/counterfactual/CounterfactualExplainerTest.java [hunk 1]
```diff
@@ -146,10 +151,16 @@ void testNonEmptyInput(int seed) throws ExecutionException, InterruptedException
 
     @ParameterizedTest
     @ValueSource(ints = { 0, 1, 2 })
+    @Timeout(value = 2, unit = TimeUnit.MINUTES, threadMode = Timeout.ThreadMode.SEPARATE_THREAD)
     void testCounterfactualMatch(int seed) throws ExecutionException, InterruptedException, TimeoutException {
+
+        System.out.println("Print something BEFORE seeding #####");
```
Comment (RobGeada) [inline explainability-core/src/test/java/org/kie/trustyai/explainability/local/counterfactual/CounterfactualExplainerTest.java]: Are these prints critical or can we remove them?

Comment (spolti) [reply explainability-core/src/test/java/org/kie/trustyai/explainability/local/counterfactual/CounterfactualExplainerTest.java]: This PR is not ready yet :)
needs a few clean up

Comment (spolti) [reply explainability-core/src/test/java/org/kie/trustyai/explainability/local/counterfactual/CounterfactualExplainerTest.java]: Done :)

**Review by RobGeada — APPROVED** [review]

**Pull request merged**